### DISCUSSION TITLE
Allow enabling/disabling Firehose tracing through HTTP API

### DISCUSSION THREAD

**Discussion by aaron-seo** [rabbitmq/rabbitmq-server#11749] [Questions]
As of today, Firehose tracing can only be enabled through `rabbitmqctl`. Curious to hear if there were previous discussions to not include it in the HTTP API? (it seems the functionality is quite old, so understandable if this context has been lost). If not and there aren't other concerns, then I'd be interested to look into a contribution to allow enabling and disabling Firehose tracing through the HTTP API.

I've seen [this previous thread](https://www.rabbitmq.com/amqp-wireshark), which suggests deprecation of the `rabbitmq_tracing` plugin. If that idea does go through for 4.x, would the built-in Firehose tracing functionality (without the plugin) be also deprecated?

**Comment by michaelklishin** [comment]
[Here are the endpoints](https://github.com/rabbitmq/rabbitmq-server/blob/main/deps/rabbitmq_tracing/src/rabbit_tracing_mgmt.erl#L14-L23) the tracing plugin uses. There are "special APIs".

To find out what HTTP methods are used you need to inspect the handler modules.

**Reply by aaron-seo** [reply]
Hi thanks for the pointer :) Before I dig into this further, are the maintainers opposed to the idea of enabling/disabling firehose tracing through the HTTP api? (it is only currently allowed through rabbitmqctl, as far as I'm aware)

**Reply by michaelklishin** [reply]
We are not opposed to it but this is the first time someone has asked for it in ≈ 11 years.

The same operations can be added as `POST /api/vhosts/{name}/tracing` and `DELETE /api/vhosts/{name}/tracing`, or something like that.
